src/main/resources/assets/computercraft/lua/rom/modules/main/http/pastebin.lua

```diff
@@ -0,0 +1,150 @@
+
+--- Attempts to guess the pastebin ID from the given code or URL
+local function extractId(paste)
+  local patterns = {
+    "^([%a%d]+)$",
+    "^https?://pastebin.com/([%a%d]+)$",
+    "^pastebin.com/([%a%d]+)$",
+    "^https?://pastebin.com/raw/([%a%d]+)$",
+    "^pastebin.com/raw/([%a%d]+)$",
+  }
+
+  for i = 1, #patterns do
+    local code = paste:match( patterns[i] )
+    if code then return code end
+  end
+
+  return nil
+end
+
+function download(url)
+  if type( url ) ~= "string" then
+    error( "bad argument #1 (expected string, got " .. type( url ) .. ")", 2 )
+  end
+
+  if not http then
+    return false, "Pastebin requires http API"
+  end
+
+  local paste = extractId( url )
```

Comment:
I think it'd probably be better to have this exposed as part of the API, `download`, `get`, etc... only accept the code.